Referee.cs

```diff
@@ -0,0 +1,1179 @@
+﻿﻿using System;
+using System.Runtime.Serialization;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEngine;
```

Comment:
Why do we need UnityEngine ? If it's only for Vector2, we could use System.Numerics, so people will be able to use some piece of your code in CodinGame directly.